### ISSUE TITLE
Adding MCUIOT to config.json prevents Homebridge from starting

### ISSUE THREAD

**Issue by Techno-Babble** [NorthernMan54/homebridge-mcuiot#5]
After adding to the config.json:
```
{
    "platform": "mcuiot",
    "name": "MCUIOT"
}
```
Homebridge will no longer successfully load.
```
systemctl status homebridge reports:
homebridge.service - Node.js HomeKit Server
Loaded: loaded (/etc/systemd/system/homebridge.service; enabled; vendor prese
Active: activating (auto-restart) (Result: exit-code) since Sat 2018-05-12 01
Process: 14608 ExecStart=/usr/bin/homebridge $HOMEBRIDGE_OPTS (code=exited, st
Main PID: 14608 (code=exited, status=1/FAILURE)
May 12 01:20:54 home-MS-7969 systemd[1]: homebridge.service: Main process exited May 12 01:20:54 home-MS-7969 systemd[1]: homebridge.service: Unit entered failed May 12 01:20:54 home-MS-7969 systemd[1]: homebridge.service: Failed with result
```

**Comment by NorthernMan54** [comment]
Unfortunately that log fragment doesn't show why homebridge crashed.  This will dump the log from systemd, and it should show the log including the error.  Also this log can be very large, please look thru it for the section with the error and share it.

sudo journalctl -b -u homebridge | cat

**Comment by Techno-Babble** [comment]
I ran the above and looked over it. There are some warnings early on in the loading process right before it loads The MCUIOT platform:
```
May 12 11:06:18 home-MS-7969 homebridge[15037]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
May 12 11:06:18 home-MS-7969 homebridge[15037]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 11:06:18 home-MS-7969 homebridge[15037]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
May 12 11:06:18 home-MS-7969 homebridge[15037]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
May 12 11:06:18 home-MS-7969 homebridge[15037]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 11:06:18 home-MS-7969 homebridge[15037]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
May 12 11:06:18 home-MS-7969 node[15037]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
May 12 11:06:18 home-MS-7969 node[15037]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 11:06:18 home-MS-7969 node[15037]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
May 12 11:06:18 home-MS-7969 node[15037]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
May 12 11:06:18 home-MS-7969 node[15037]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 11:06:18 home-MS-7969 node[15037]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
May 12 11:06:18 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:18] Loaded plugin: homebridge-mcuiot
May 12 11:06:18 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:18] Registering platform 'homebridge-mcuiot.mcuiot'
```
Then this is the log leading up to the error:
```
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] [MCUIOT] Starting mDNS listener
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] [Config] Console is listening on port xxxx.
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] Homebridge is running on port xxxxx.
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] [MCUIOT] Web Server listening on: http://localhost:xxxx
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] [MCUIOT] Found MCUIOT device: NODE-D5E6XX
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] [TP-Link] New Device Online: [Fish Tank] plug [8006FEA6E492DB899866490E13D59F6D175BXCXC] 192.168.1.221 xxxx
May 12 11:06:19 home-MS-7969 homebridge[15037]: [2018-5-12 11:06:19] [TP-Link] Adding: [Fish Tank] plug [8006FEA6E492DB899866490E13D59F6D175BXCXC]
May 12 11:06:19 home-MS-7969 systemd[1]: homebridge.service: Main process exited, code=exited, status=1/FAILURE
May 12 11:06:19 home-MS-7969 systemd[1]: homebridge.service: Unit entered failed state.
May 12 11:06:19 home-MS-7969 systemd[1]: homebridge.service: Failed with result 'exit-code'.
May 12 11:06:29 home-MS-7969 systemd[1]: homebridge.service: Service hold-off time over, scheduling restart.
May 12 11:06:29 home-MS-7969 systemd[1]: Stopped Node.js HomeKit Server.
```

**Comment by NorthernMan54** [comment]
The warnings for avahi are normal and not an issue, the rest of the log you shared unfortunately does not include the actual error causing homebridge to exit.  Can you log at the log further, and maybe try again.

Or you could try running homebridge from the command line.

**Comment by Techno-Babble** [comment]
Here is a full run attempt from that log. I’ll attempt to run from the command line as well.
```
May 12 12:00:52 home-MS-7969 systemd[1]: homebridge.service: Service hold-off time over, scheduling restart.
May 12 12:00:52 home-MS-7969 systemd[1]: Stopped Node.js HomeKit Server.
May 12 12:00:52 home-MS-7969 systemd[1]: Started Node.js HomeKit Server.
May 12 12:00:53 home-MS-7969 homebridge[31972]: ┌──────────────────────────────────────────────────────────────┐
May 12 12:00:53 home-MS-7969 homebridge[31972]: │                   npm update check failed                    │
May 12 12:00:53 home-MS-7969 homebridge[31972]: │             Try running with sudo or get access              │
May 12 12:00:53 home-MS-7969 homebridge[31972]: │             to the local update config store via             │
May 12 12:00:53 home-MS-7969 homebridge[31972]: │ sudo chown -R $USER:$(id -gn $USER) /home/homebridge/.config │
May 12 12:00:53 home-MS-7969 homebridge[31972]: └──────────────────────────────────────────────────────────────┘
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loaded plugin: homebridge-config-ui
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Registering platform 'homebridge-config-ui.config'
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] ---
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loaded plugin: homebridge-ecoplug
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Registering platform 'homebridge-ecoplugs.EcoPlug'
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] ---
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loaded plugin: homebridge-hs100
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Registering platform 'homebridge-hs100.Hs100'
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] ---
May 12 12:00:53 home-MS-7969 homebridge[31972]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
May 12 12:00:53 home-MS-7969 homebridge[31972]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 12:00:53 home-MS-7969 homebridge[31972]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
May 12 12:00:53 home-MS-7969 homebridge[31972]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
May 12 12:00:53 home-MS-7969 homebridge[31972]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 12:00:53 home-MS-7969 homebridge[31972]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
May 12 12:00:53 home-MS-7969 node[31972]: *** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
May 12 12:00:53 home-MS-7969 node[31972]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 12:00:53 home-MS-7969 node[31972]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
May 12 12:00:53 home-MS-7969 node[31972]: *** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
May 12 12:00:53 home-MS-7969 node[31972]: *** WARNING *** Please fix your application to use the native API of Avahi!
May 12 12:00:53 home-MS-7969 node[31972]: *** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loaded plugin: homebridge-mcuiot
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Registering platform 'homebridge-mcuiot.mcuiot'
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] ---
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loaded plugin: homebridge-platform-wemo
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Registering platform 'homebridge-platform-wemo.BelkinWeMo'
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] ---
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loaded config.json with 0 accessories and 5 platforms.
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] ---
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] Loading 5 platforms...
May 12 12:00:53 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:53] [Config] Initializing config platform...
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [TP-Link] Initializing Hs100 platform...
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [TP-Link] homebridge-hs100 v3.7.0, node v8.11.1, homebridge v0.4.42
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [EcoPlug] Initializing EcoPlug platform...
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [WeMo Platform] Initializing BelkinWeMo platform...
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [MCUIOT] Initializing mcuiot platform...
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] Loading 0 accessories...
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [TP-Link] Configuring cached accessory: [Fish Tank] 8006FEA6E492DB899866490E13D59F6D175BXCXC ce7f0e62-6bcc-48f7-ba86-62314219941b
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [EcoPlug] configureAccessory ECO-7806F52D Deck Lights
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [MCUIOT] configureAccessory MCUIOT Reset Switch
May 12 12:00:54 home-MS-7969 homebridge[31972]: Setup Payload:
May 12 12:00:54 home-MS-7969 homebridge[31972]: X-HM://0023ISYWY5IG1
May 12 12:00:54 home-MS-7969 homebridge[31972]: Scan this code with your HomeKit app on your iOS device to pair with Homebridge:
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]:                                                       
May 12 12:00:54 home-MS-7969 homebridge[31972]: Or enter this code with your HomeKit app on your iOS device to pair with Homebridge:
May 12 12:00:54 home-MS-7969 homebridge[31972]:                        
May 12 12:00:54 home-MS-7969 homebridge[31972]:     ┌────────────┐
May 12 12:00:54 home-MS-7969 homebridge[31972]:     │ xxx-xx-xxx │
May 12 12:00:54 home-MS-7969 homebridge[31972]:     └────────────┘
May 12 12:00:54 home-MS-7969 homebridge[31972]:                        
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [MCUIOT] Starting mDNS listener
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [Config] Console is listening on port xxxx.
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] Homebridge is running on port xxxxx.
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [MCUIOT] Web Server listening on: http://localhost:xxxx
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [MCUIOT] Found MCUIOT device: NODE-D5E6XX
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [TP-Link] New Device Online: [Fish Tank] plug [8006FEA6E492DB899866490E13D59F6D175BXCXC] 192.168.1.221 XXXX
May 12 12:00:54 home-MS-7969 homebridge[31972]: [2018-5-12 12:00:54] [TP-Link] Adding: [Fish Tank] plug [8006FEA6E492DB899866490E13D59F6D175BXCXC]
May 12 12:00:54 home-MS-7969 homebridge[31972]: /usr/lib/node_modules/homebridge-mcuiot/index.js:466
May 12 12:00:54 home-MS-7969 systemd[1]: homebridge.service: Main process exited, code=exited, status=1/FAILURE
May 12 12:00:54 home-MS-7969 systemd[1]: homebridge.service: Unit entered failed state.
May 12 12:00:54 home-MS-7969 systemd[1]: homebridge.service: Failed with result 'exit-code'.
```

**Comment by NorthernMan54** [comment]
If you look at at the last log the error starts then it switches to the systemd stuff at the very end, about 5 lines up

Need a little bit more detail, line 466 is something with name aliases

**Comment by Techno-Babble** [comment]
Here is the error from running it in the command line:
```
/usr/lib/node_modules/homebridge-mcuiot/index.js:466
    var displayName = this.aliases[name];
                                  ^

TypeError: Cannot read property 'NODE-D5E6AA' of undefined
    at mcuiot.addMcuAccessory (/usr/lib/node_modules/homebridge-mcuiot/index.js:466:35)
    at /usr/lib/node_modules/homebridge-mcuiot/index.js:147:16
    at Object.<anonymous> (/usr/lib/node_modules/homebridge-mcuiot/index.js:447:7)
    at Request._callback (/usr/lib/node_modules/homebridge-mcuiot/index.js:747:7)
    at Request.self.callback (/usr/lib/node_modules/homebridge-mcuiot/node_modules/request/request.js:186:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/usr/lib/node_modules/homebridge-mcuiot/node_modules/request/request.js:1163:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/usr/lib/node_modules/homebridge-mcuiot/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:313:30)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11) 
```

**Comment by NorthernMan54** [comment]
Looks like I have a problem with the handling of aliases if you don’t have any at all.  As a quick fix can you add an alias section to the config?  It should fix the problem

**Comment by Techno-Babble** [comment]
Adding an alias for the node fixed the issue. It’s up and running now.

**Comment by NorthernMan54** [comment]
Fixed in release v0.1.50
